### ISSUE TITLE
macOS: 1.2 builds use lots of battery

### ISSUE THREAD

**Issue by aronatkins** [rstudio/rstudio#3757]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.2.1074
    OS Version      : macOS 10.14
    R Version       : 3.5.1

### Steps to reproduce the problem

Download the latest 1.2 preview build on macOS. Unplug your laptop. Launch the RStudio IDE. Launch the Activity Monitor. Wait a while. Watch your battery drain quickly.

I do have a few other things open (Safari, Emacs, iTerm2), but nothing else is using much power. I do _not_ have Chrome running, as it is a similar battery hog.

### Describe the problem in detail

Woah!

<img width="645" alt="image" src="https://user-images.githubusercontent.com/362187/47577271-6f9e0380-d914-11e8-9bff-7f9d2f220655.png">

Egads!

<img width="173" alt="image" src="https://user-images.githubusercontent.com/362187/47577350-ae33be00-d914-11e8-8f21-50024e54c376.png">

### Describe the behavior you expected

Older RStudio macOS builds were pretty friendly to battery life. That is no longer the case. Using the IDE seriously alters my time-unplugged.

**Comment by jjallaire** [comment]
Once we can repro this we might want to binary-search over the 1.2 series of builds to see where it was introduced. We do see this sort of thing happen from time-to-time, and it's usually because there is a JS or Qt codepath we introduce that is either or both unexpectedly expensive and frequent.

**Comment by kevinushey** [comment]
Looks like this is caused by leaked timers from our progress panels. We schedule an animation for our progress panels (using a spinner) here:

https://github.com/rstudio/rstudio/blob/93a7722965e3f46a2b5695b2d609cd10d2fee9db/src/gwt/src/org/rstudio/core/client/widget/ProgressSpinner.java#L73-L84

Previously, these were only used for chunk output widgets, but in v1.2 we started using them for the progress panels as well:

https://github.com/rstudio/rstudio/commit/6e70ebd179eebde3eefa927dad2908d2e1b06abd#diff-3a3939b13b6810dc86fe8e005a95a5cc

However, the animation used for these panels is only stopped when one calls `detach()`, and this never happened for these spinners -- effectively leaking the animation loop. The solution will likely involve figuring out the right place to call `detach()` and calling it where needed.

**Comment by jmcphers** [comment]
See also https://github.com/rstudio/rstudio/commit/0667312cea7e16230a21170674129f7e3478e2ac#diff-3a3939b13b6810dc86fe8e005a95a5cc -- we shouldn't be running the animation when the panels aren't visible (possibly the panels think they're visible?)

**Comment by kevinushey** [comment]
You're right, the animations themselves are not redrawing -- however, the command that schedules the redraw (checking `isVisible()` and bumping the frame count) is indeed being leaked, so every ProgressPanel generated effectively leaks a `setInterval()` timer that runs every 75ms.

**Comment by rich-rstudio** [comment]
verified on 1.2.1090
